style_guide/README.md

```diff
@@ -1,206 +1,307 @@
 - Machine names
 - Proprietary data
 
-If `imap_processing` code needs access to this information, it should be stored
-in a configuration file that is not part of the repository.
+If `imap_processing` code needs access to this information, it should be stored in a configuration file that is not part
+of the repository.
 
 
 ## Naming Conventions
 
-TBD
+*To be determined*
+
 
 ## Tools and Library Recommendations
 
+- `black` for checking python code formatting
+- `pytest` for unit testing
+- `ruff` for checking docstring formatting
 - `spiceypy` for using SPICE kernels
 - `space-packet-parser` to unpack CCSDS packets
 
-## Releases
 
-TBD
+## Versioning
+
+Any changes pushed to the `main` branch should be tagged with a version number. The version number convention is
+`x.y.z`, where
+
+    x = The main version number.  Increase when making incompatible API changes.
+    y = The feature number.  Increase when change contains a new feature with or without bug fixes.
+    z = The hotfix number. Increase when change only contains bug fixes.
+
+
+## Release Workflow
+
+Any changes merged or pushed to the `main` branch should follow a specific release workflow which follows the diagram
+shown in the [git & GitHub Workflow](#git-&amp-github-workflow) section. This includes both 'nominal releases' (i.e. new
+features being merged into `main`) and 'hotfixes' (i.e. bug fixes made directly to `main`). These workflows are
+described below. It is assumed that steps (1) through (4) in the [git & GitHub Workflow](#git-&amp-github-workflow) section
+are already completed.
+
+### Nominal releases
+
+1. Make sure the `dev` branch is up-to-date with any changes you want included in the release (i.e. merge in any feature
+   branches using the nominal [git/GitHub workflow](#git-&amp-github-workflow)).
+2. Create a new 'release' branch off of `dev`.  The name of the branch should match the version number to be used for
+   the release, which should follow the [versioning](#versioning) conventions.
+3. Make any release-specific commits to the new release branch using the nominal `git add`/`git commit` cycle. This may
+   include commits that add release notes, or update version numbers in various configurations.
+4. Push the release branch to the main `IMAP-Science-Operations-Center` `imap_processing` repo (i.e. `upstream`).
+5. In GitHub, create two pull requests: one that merges the release branch into `main`, and one that merges the release
+   branch into `dev`. Proceed with the nominal review & merge process described in steps (10) and (11) in the 
+6. proceed with the nominal review & merge process described in steps (10) and (11) in the
+   [git/GitHub workflow](#git-&amp-github-workflow) section.
+7. Once the changes are merged into the `main` branch, create a
+   [new release](https://github.com/IMAP-Science-Operations-Center/imap_processing/releases) for the merge commit and
+   assign the appropriate version number.
+
+### Making hotfixes
+
```

Comment:
It might be worth adding a note that the hotfix flow should NOT be used unless it's time sensitive. This may or many not be necessary to clarify, probably most people know what a hotfix is
```suggestion

A hotfix should not be created unless there is some specific reason to bypass the normal `dev` branch workflow.

```